### PR TITLE
[Dashboard] Improve faucet claim

### PR DESCRIPTION
**Pull request by kien-ngo**
## Problem solved

Short description of the bug fixed or feature added

---

## PR-Codex overview
This PR focuses on enhancing the functionality of the `Faucet` feature in the dashboard application. It introduces new environment variables, updates modal components to allow closing, and modifies API routes to ensure email verification for faucet claims.

### Detailed summary
- Added `REDIS_URL` to `.env.example`.
- Modified `turbo.json` to include new environment variables.
- Updated `OnboardingModal` to accept `onOpenChange` prop.
- Enhanced `FaucetButton` to handle email verification.
- Updated API routes to check for verified email before allowing claims.
- Introduced caching for account claims based on IP and account ID.

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by kien-ngo** [comment]
* **#5226** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/5226?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @kien-ngo and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: apps/dashboard/src/app/api/testnet-faucet/claim/route.ts [hunk 1]
```diff
@@ -19,6 +21,49 @@ interface RequestTestnetFundsPayload {
...
+  if (accountRes.status !== 200) {
+    // Account not found on this connected address
+    return NextResponse.json(
+      {
+        error: "thirdweb account not found",
+      },
+      { status: 400 },
+    );
+  }
+  console.log(await accountRes.json());
```
[the remark this answers is not in the document]
Comment (kien-ngo) [reply apps/dashboard/src/app/api/testnet-faucet/claim/route.ts]: I love this bot

File: apps/dashboard/src/app/(dashboard)/(chain)/[chain_id]/(chainPage)/components/client/FaucetButton.tsx [hunk 2]
```diff
@@ -1,5 +1,6 @@
 "use client";
 
+import { ChakraProviderSetup } from "@/components/ChakraProviderSetup";
```
Comment (gregfromstl) [inline apps/dashboard/src/app/(dashboard)/(chain)/[chain_id]/(chainPage)/components/client/FaucetButton.tsx:3]: Chakra? I thought we were done with Chakra?

Comment (kien-ngo) [reply apps/dashboard/src/app/(dashboard)/(chain)/[chain_id]/(chainPage)/components/client/FaucetButton.tsx:3]: Not even close lol. Might take another few weeks

**Review by MananTank — APPROVED** [review]

**Review by jnsdls — APPROVED** [review]

**Pull request merged**